### PR TITLE
added class search

### PR DESCRIPTION
**Pull request by dwee9**

### CODE REVIEW

File: iOS/Views/TableCells/ClassGroupCell.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            ImageDownloader.GetImageForURL( course.PhotoURL, ( image ) =>
+             {
+                 imageview.Image = image;
+             } );
+            nameLabel.Text = course.Name;
+            subLabel.Text = course.MeetingTime;
+
+            CLLocation coursesLocation = new CLLocation( course.Latitude, course.Longitude );
+            var distance = LocationManager.Instance.currentLocation.DistanceFrom( coursesLocation ) / 1609;
```
Comment (jhawkzz) [inline iOS/Views/TableCells/ClassGroupCell.cs]: Could we define 1609 as a `MetersToMiles` constant?

File: iOS/Views/CollectionCells/ClassGroupCollectionCell.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        public void instantiateWithCourse( MobileAppCourse course )
+        {
+            this.course = course;
+            ImageDownloader.GetImageForURL( course.PhotoURL, ( image ) =>
+            {
+                imageview.Image = image;
+            } );
+            nameLabel.Text = (course.Name ?? "").ToUpper( );
+            CLLocation campusLocation = new CLLocation( course.Latitude, course.Longitude );
+            var distance = LocationManager.Instance.currentLocation.DistanceFrom( campusLocation ) / 1609;
```
Comment (jhawkzz) [inline iOS/Views/CollectionCells/ClassGroupCollectionCell.cs]: Could we define 1609 as a `MetersToMiles` constant?

File: iOS/ViewControllers/Class/ClassSearchViewController.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            classNameField.setDarkKeyboard( );
+            classKeywordsField.setDarkKeyboard( );
+            zipField.textField.KeyboardType = UIKeyboardType.NumberPad;
+            this.addDoneButtonToKeyboard( zipField.textField );
+        }
+
+        partial void searchAction( Foundation.NSObject sender )
+        {
+            UIBlocking.Instance.start( );
+            CourseController.SearchCourses( 0, 10,
```
Comment (jhawkzz) [inline iOS/ViewControllers/Class/ClassSearchViewController.cs]: The current app offers a "Tap for 10 more" row at the bottom that loads the next 10 classes. Pretty sure we'll need that for both Courses and Groups in the search results list

File: iOS/Views/TableCells/ClassGroupCell.cs (hunk 1 above)
Comment (dwee9) [reply iOS/Views/TableCells/ClassGroupCell.cs]: Sure, do you want it in the app.shared or just in the controller?

File: iOS/ViewControllers/Class/ClassSearchViewController.cs (hunk 3 above)
Comment (dwee9) [reply iOS/ViewControllers/Class/ClassSearchViewController.cs]: Yeah, that is in a separate ticket :)

Comment (jhawkzz) [reply iOS/ViewControllers/Class/ClassSearchViewController.cs]: Perfect! Sorry, I wasn't sure so I thought I'd ask :)

File: iOS/Views/TableCells/ClassGroupCell.cs (hunk 1 above)
Comment (jhawkzz) [reply iOS/Views/TableCells/ClassGroupCell.cs]: Don't have a preference - I guess wherever you want, and if we need to move it for Android we can

**Pull request merged**